var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -218,8 +218,14 @@ class Paraview(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("tbb", when="+tbb")
 
     depends_on("mpi", when="+mpi")
-    depends_on("qt+opengl", when="@5.3.0:+qt+opengl2")
-    depends_on("qt~opengl", when="@5.3.0:+qt~opengl2")
+    depends_on("qmake", when="@5.12.0:+qt")
```

Comment:
The virtual `qmake` is a really odd way to express providing Qt API. Especially for something like ParaView, and most other projects using Qt, which don't use any qmake.

I think maybe `qt` and `qt-base` should be merged using different `build_system` definitions for the two versions.

https://spack.readthedocs.io/en/latest/packaging_guide.html#multiple-build-systems

I see you are the maintainer for the QtBase. Is that something you think you could do?

Reply:
The Qt5 package is a monolithic package that is difficult to maintain. With Qt6 we consciously started a new package because it allows us to have downstream packages depend on the components they need, and only on `qt-base` if only the core is needed. Merging `qt` and `qt-base` would remove this advantage. (@sethrj FYI, in case you wish to comment.)

I do agree that `qmake` is maybe not the best virtual name.

Reply:
I see. Yeah digging a bit more into the archive I see patches for multiple components Qt, qtbase being only one. 

If we can get a better virtual name that would be ideal. If `qt-base` is what we actually want for most applications, something like `qt-base-api` like we did for `zlib-api` and have both provide that.

After QT5 is EOL maybe we convert Qt to a `BundlePackage`, that depends on all off the Qt components in spack and drop the virtuals. That way people can say "give me everything from Qt" easily.

Reply:
Yes, `qt-base-api` is a better name (but there is not really a uniform API between Qt5 and Qt6).

As for Qt5 EOL, we are still carrying Qt4 and Qt3 versions (without deprecation), so that might still be a while before there are no users...

> could we add the 5.15 versions to the `qt-base` package now?

I don't understand this. We have 5.15 in `qt`. The build system for Qt6 is different from Qt5, so we cannot simply drop the 5.15 versions in `qt-base` (the Qt6 base package).

In the context of this PR, `paraview` still supports the same Qt5 versions as before; that's unchanged (or at least that's the intention, if I didn't screw anything up).

Reply:
With virtual packages, you can define a version. We do this for LLVM, because like Qt the API is not stable between major release and we have many providers.

So Qt 5.15.x would `provide("qt-base-api@5.15")` and Qt 6.7 would provide `provide("qt-base-api@6.7")`. You can then depend on `qt-base-api@X.Y` or use the `[virtuals=qt-base-api] qt-base@6.7`